documentation/Using-OVS-DPDK-and-COR.md

```diff
@@ -0,0 +1,82 @@
+```
+
+Next we will set up DPDK:
+```
+git clone http://dpdk.org/git/dpdk -b v16.11 $HOME/dpdk
+cd $HOME/dpdk
+export RTE_SDK=$(pwd)
+export RTE_TARGET=x86_64-native-linuxapp-gcc
+make -j install T=$RTE_TARGET DESTDIR=install EXTRA_CFLAGS='-g'
+sudo cp -f install/lib/lib* /lib64/
+export DPDK_BUILD_DIR=x86_64-native-linuxapp-gcc
+```
+Once this completes with out issue, build and configure OVS:
+```
+git clone https://github.com/openvswitch/ovs -b branch-2.6 $HOME/ovs
+cd $HOME/ovs
+./boot.sh
+./configure --prefix=/usr --sysconfdir=/etc --localstatedir=/var \
+    --with-dpdk=$RTE_SDK/$DPDK_BUILD_DIR  --disable-ssl --with-debug  CFLAGS='-g'
+sudo -E make install -j
+```
+
+## Running OVS-DPDK on host system
+Once installed without error, we'll need to start up the ovsdb-server and ovs-vswitchd.  The Open_vSwitch configuration settings will vary depending on the hardware you are running on (ex: lcore-mask, socket-mem).
+```
+sudo mkdir -p /var/run/openvswitch
+sudo killall ovsdb-server ovs-vswitchd
+rm -f /var/run/openvswitch/vhost-user*
+sudo rm -f /etc/openvswitch/conf.db
+
+export DB_SOCK=/var/run/openvswitch/db.sock
+sudo -E ovsdb-tool create /etc/openvswitch/conf.db /usr/share/openvswitch/vswitch.ovsschema
+sudo -E ovsdb-server --remote=punix:$DB_SOCK --remote=db:Open_vSwitch,Open_vSwitch,manager_options --pidfile --detach
+
+sudo -E ovs-vsctl --no-wait init
+sudo -E ovs-vsctl --no-wait set Open_vSwitch . other_config:dpdk-lcore-mask=0x3
+sudo -E ovs-vsctl --no-wait set Open_vSwitch . other_config:dpdk-socket-mem=1024
+sudo -E ovs-vsctl --no-wait set Open_vSwitch . other_config:dpdk-init=true
+
+sudo sysctl -w vm.nr_hugepages=1024
+sudo mount -t hugetlbfs -o pagesize=2048k none /dev/hugepages
+
+sudo -E ovs-vswitchd unix:$DB_SOCK --pidfile --detach --log-file=/var/log/openvswitch/ovs-vswitchd.log
+ps -ae | grep ovs
+```
+At this point your host system is ready to start creating DPDK enabled OVS switches.
+
+## Grab and install OVSDPDK  Docker plugin
+We need to install an OVS-DPDK Docker plugin in order to facilitate creating a network and connecting clear-containers to this network via Docker.
+
+Details on pulling and installing the plugin can be found at https://github.com/clearcontainers/ovsdpdk
+
+## Example: Launching two Clear Containers using OVS-DPDK
+
+To make use of OVS-DPDK, the next step is to use Docker to create a network using the OVS-DPDK switch.  An example:
+```
+sudo docker network create -d=ovsdpdk --ipam-driver=ovsdpdk --subnet=192.168.1.0/24 --gateway=192.168.1.1  --opt "bridge"="ovsbr" ovsdpdk_net
+```
+You can verify the OVS switch was created by looking for the ovsbr in the output of:
+```
+sudo ovs-vsctl show
+```
+
+You can now test connectivity by launching two containers as follows (this assumes that you have Docker setup to use this branch's runtime):
+```
+sudo docker run -d --net=ovsdpdk_net --ip=192.168.1.2 --mac-address=CA:FE:CA:FE:01:02 --name "ccovs1" debian bash -c "ip a; ip route; sleep 30000"
```

Comment:
I've been using something like `tail -f /dev/null` for the same effect - except it is 'forever'. Just a thought, rather than having a notional 'forever' value (well, 30000 seconds is a bit over 8h I think, but you see what I mean :-) )